qiskit/aqua/algorithms/phase_estimators/phase_estimator.py

```diff
@@ -0,0 +1,213 @@
+"""The Quantum Phase Estimation Algorithm."""
+
+from typing import Dict, Optional, Union
+from qiskit.circuit import QuantumCircuit
+import qiskit
+#from qiskit.result import Counts
+from qiskit.circuit.library import PhaseEstimation
+from qiskit.circuit.classicalregister import ClassicalRegister
+from qiskit.providers import BaseBackend
+from qiskit.aqua import QuantumInstance
+from qiskit.aqua.algorithms import QuantumAlgorithm
+from qiskit import execute
+from qiskit.aqua.utils import get_subsystem_density_matrix
+import numpy as np
+
+
+class PhaseEstimator(QuantumAlgorithm):
+    """The Quantum Phase Estimation algorithm.
+    """
+
+    def __init__(self,
+                 num_evaluation_qubits,
+                 pe_circuit = None,
+                 num_unitary_qubits = None,
+                 unitary = None,
+                 input_state_circuit = None,
```

Comment:
Can we call this `state_preparation`? Then it matches with the Grover operator and Grover algorithm. In QAE this all comes together so it would be nice if the same things have the same name.

Reply:
* type hints. Yes, I was leaving this till the inputs and outputs of the methods had settled.
* docstrings. I tried to abstract standards by reading existing docstrings, but was unable; they seemed to vary. The link you provided says the document does not exist: "This shared file or folder link has been removed or is unavailable to you."  I suspect it's a problem on my end. I'm not using vpn on this browser. But, I can see other boxnotes.
* Results object. I think the pedagogical uses of a QPE class are broader than just returning the most likely string. Sometimes the most likely string is not of interest. But, maybe computing and storing the most likely string regardless of whether it's needed is ok.